packages/filters/src/Filter.tsx

```diff
@@ -33,14 +34,24 @@ const Filter = <T, K extends keyof T>({
     updateFilter(filterKey as string, serializedValue, value)
   }
 
+  // If no filter key is provided,
+  // filterKey provided to component will be used
+  const _removeFilter = (fkey?: K) => {
+    removeFilter(fkey ?? filterKey)
+  }
+
   // Retrieves the raw and parsed filter value
   const rawValue = getFilterValue(filterKey as string)
   const parsedValue = rawValue as T
 
   // Render the children with the injected props
   return (
     <div id="filter" className={className}>
-      {children({ onChange: handleChange, value: parsedValue as T[K], removeFilter })}
+      {children({
+        onChange: handleChange,
+        value: parsedValue as T[K],
+        removeFilter: _removeFilter
```

Comment:
Just add a inline function here, instead of creating _removeFilter. 
_removeFilter is not readable. You can rename the method too -- wrappedRemoveFilter

```suggestion
        removeFilter: (key: k) => removeFilter(key || filterKey)
```

Reply:
I will rename the method, would be more readable as a separate function I guess